client/dfclient/src/gameplay_state.cpp

```diff
@@ -269,13 +280,23 @@ void GameplayState::OnMessage(const std::string& data) {
 			// this is the first time we see this mob, create it
 			Mob newMob;
 			newMob.sprite = CreateNewAnimSprite(this->cameraNode.get(), mobData->species());
+			newMob.sprite->setAlpha(1);
+			//fade in
+			_resources._mobTweens[mobData->mobID()] = std::move(CreateAlphaTransitionTween(newMob.sprite.get(), 1, 255, MOB_FADEIN_TIME));
```

Comment:
`std::move` is completely unnecessary here. It would make sense if you moved out of a variable, but in this case move assignment operator will be called even without the `std::move`.

BTW, if you would like to learn more about how moving and rvalue references work, this is an article which makes it clear (for me at least): http://thbecker.net/articles/rvalue_references/section_01.html

Reply:
Thanks, that's right. I know how it works, I just retarded here a little bit :D